homeassistant/components/environment_canada/weather.py

```diff
@@ -232,19 +232,20 @@ def get_forecast(ec_data, forecast_type):
             )
 
     elif forecast_type == "hourly":
-        hours = ec_data.hourly_forecasts
-        for hour in range(0, 24):
+        for hour in ec_data.hourly_forecasts:
             forecast_array.append(
                 {
-                    ATTR_FORECAST_TIME: dt.as_local(
-                        datetime.datetime.strptime(hours[hour]["period"], "%Y%m%d%H%M")
-                    ).isoformat(),
-                    ATTR_FORECAST_TEMP: int(hours[hour]["temperature"]),
+                    ATTR_FORECAST_TIME: datetime.datetime.strptime(
```

Comment:
Can we use this util?
https://github.com/home-assistant/core/blob/b7c1df7864468ddc4b9a2bec51894c7e25fb0104/homeassistant/util/dt.py#L124

Reply:
Unfortunately not, since the timestamp isn’t formatted in the way it expects. I looked at using dateutil, but it’s not currently part of the project.

Reply:
How is the timestamp formatted. Please give an example.

Reply:
They're UTC times expressed like this:`20210617113000`.

Reply:
Ok. If the string includes seconds we need to update the time formatting string below.